src/Plugin/GraphQL/Fields/FieldPluginBase.php

```diff
@@ -167,9 +164,22 @@ protected function resolveDeferred(callable $callback, $value, array $args, Reso
     }
 
     if (is_callable($result)) {
-      return new Deferred(function () use ($result, $value, $args, $context, $info) {
-        return $this->resolveDeferred($result, $value, $args, $context, $info);
-      });
+      return new Deferred(
+        function () use ($result, $value, $args, $context, $info) {
+          if ($this->isLanguageAwareField()) {
```

Comment:
No need to call this twice (isLanguageAwareField()). Just cache the result and re-use it here.